CalibPPS/ESProducers/python/ctppsBeamParametersFromLHCInfoESSource_cff.py

```diff
@@ -0,0 +1,7 @@
+from CalibPPS.ESProducers.ctppsBeamParametersFromLHCInfoESSource_cfi import *
+
+from Configuration.Eras.Modifier_run3_common_cff import run3_common
+run3_common.toModify(ctppsBeamParametersFromLHCInfoESSource, useNewLHCInfo = True)
```

Comment:
Python objects should be modified only in the file they are defined in (otherwise various kind of weird behavior can occur in some cases). Simplest change would be

```py
from CalibPPS.ESProducers.ctppsBeamParametersFromLHCInfoESSource_cfi import ctppsBeamParametersFromLHCInfoESSource as _ctppsBeamParametersFromLHCInfoESSource
ctppsBeamParametersFromLHCInfoESSource = _ctppsBeamParametersFromLHCInfoESSource.clone()
```
but then there would be a conflict if both the `ctppsBeamParametersFromLHCInfoESSource_cfi` and `ctppsBeamParametersFromLHCInfoESSource_cff` were loaded into the same `cms.Process`. 

The `ctppsBeamParametersFromLHCInfoESSource_cfi` is generated by `fillDescriptions()`, right? Then an established pattern would be to generate the `cfi` under e.g. the name `ctppsBeamParametersFromLHCInfoESSourceDefault_cfi`, and do here
```py
from CalibPPS.ESProducers.ctppsBeamParametersFromLHCInfoESSourceDefault_cfi import ctppsBeamParametersFromLHCInfoESSourceDefault as _ctppsBeamParametersFromLHCInfoESSourceDefault
ctppsBeamParametersFromLHCInfoESSource = _ctppsBeamParametersFromLHCInfoESSourceDefault.clone()
```
Then this file could be named as `..._cfi.py`, and all the downstream could keep importing the `cfi` file.

Reply:
> [#42515 (comment)](https://github.com/cms-sw/cmssw/pull/42515#discussion_r1297170431)
> `_cff` files that import `*` from `cfi` and use `run3_common.toModify()` have already been used in a merged PR #42515 and seem to cause no issue (https://github.com/cms-sw/cmssw/pull/42515/files#diff-cba86682cb8fea67d97d34b83f8883127a6c0f4f7b6d7454bccb7b4e29b43169)

This pattern is against our policy for the python fragment files (see 6.16 in https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1, also 6.13 and 6.14 are related). It may work today, but is brittle, and may easily break in the future.

Reply:
> I'm sorry but that comment seems to be in contrary to @perrotta's suggestion on how to handle those those modifiers - [#42515 (comment)](https://github.com/cms-sw/cmssw/pull/42515#discussion_r1297170431)

In doubt, always trust what @makortel says :-)

However, the two comments seem rather similar to me, besides the names given to the configuration files:
- I suggested to modify centrally a file named ctppsInterpolatedOpticalFunctionsESSource_cff.py that derives from a fillDescription generated ctppsInterpolatedOpticalFunctionsESSource_cfi.py
- Matti suggests to  modify centrally a file named ctppsInterpolatedOpticalFunctionsESSource_cfi.py that derives from a fillDescription generated ctppsBeamParametersFromLHCInfoESSourceDefault_cfi.py

Matti's suggested naming conventions are probably safer, in the sense that if implemented there is no risk that someone imports the unmodified fillDescription generated config thinking it is the already modified one instead.

Reply:
> Matti's suggested naming conventions are probably safer, in the sense that if implemented there is no risk that someone imports the unmodified fillDescription generated config thinking it is the already modified one instead.

Right. The important part is that the file where the Modifiers are applied `clone()`s the python object. The `clone()` forms the definition of the python object, and then (by policy) the file is allowed to customize the object.

The pattern of `fillDescriptions()`-generated `ctppsInterpolatedOpticalFunctionsESSource_cfi.py`, from where `ctppsInterpolatedOpticalFunctionsESSource` is imported from and cloned in `ctppsInterpolatedOpticalFunctionsESSource_cff.py` would be ok for the policy. If someone would import/load the `cfi` in addition to the `cff` being used elsewhere, that would at least lead to framework raising an exception already during the python processing (of two modules attempting to use the same label).

With `fillDescriptions()` generating `<name>Default_cfi`, from where `<name>` is imported and `clone()`d in `<name>_cfi`, the probability for someone importing the `<name>Default_cfi` should be smaller. If both the `<name>Default_cfi` and `<name>_cfi` would be loaded into the same `cms.Process`, there would be no label collision, and the further behavior depends on the modules in question (for ES modules framework should throw an exception about two ES modules trying to produce the same things).

Reply:
I hope commit 4b795e5c13780fe4450986dbb9732f7d20b1826c  addresses this issue and makes the code a bit safer to use :)
Please take a look and let me know what do you think - if the code looks fine I'll merge the 2 commits

Reply:
Btw I had to create 2 `fillDescription` methods for my needs and did so using only my intuition - any advice would be welcome

Reply:
> Btw I had to create 2 `fillDescription` methods for my needs and did so using only my intuition - any advice would be welcome

Do you mean having one in `CTPPSProtonReconstructionEfficiencyEstimatorMC` and another in `CTPPSProtonReconstructionSimulationValidator`?